lib/msf/core/analyze/result.rb

```diff
@@ -29,7 +29,51 @@ def evaluate(with: @datastore, payloads: @wanted_payloads)
     self
   end
 
-  def to_s
+  # Returns state for module readiness.
+  #
+  # @return :sym the stateful result one of:
+  #  * :READY_FOR_TEST, :REQUIRES_CRED, :REUSE_PREVIOUS_OPTIONS, :MISSING_REQUIRED_OPTION, :MISSING_PAYLOAD, :REQUIRES_SESSION, :NEEDS_TARGET_ACTION, :INVALID_OPTION, :NOT_APPLICABLE
+  #
+  # | State                   | Detailed Reason                                                                                                                                                                |
+  # |-------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+  # | READY_FOR_TEST          | Ready for Test - All required options have defaults                                                                                                                            |
+  # | REQUIRES_CRED           | Requires DB Credentials - Required options have defaults except credential values - if db contains known credentials for required fields validation is possible                |
+  # | REUSE_PREVIOUS_OPTIONS  | Reuse Previous Options-  Taken as an analysis option, process existing module runs to gather options set for same module on other hosts                                        |
+  # | MISSING_REQUIRED_OPTION | Missing Required Options - Some options are not available requiring manual configuration                                                                                       |
+  # | MISSING_PAYLOAD         | Missing Compatible Payload - Known host details and payload restrictions exclude all payloads                                                                                  |
+  # | REQUIRES_SESSION        | Requires Session - Modules that require an existing session can cannot be executed as first entry point on targets                                                             |
+  # | NEEDS_TARGET_ACTION     | Needs target action - Module that either start a service and need the target to respond in a way that may require user interaction. (Browser exploit, needs target reboot....) |
+  # | INVALID_OPTION          | Options used in Result evaluation are invalid                                                                                                                                  |
+  # | NOT_APPLICABLE          | Module is not applicable to the host                                                                                                                                           |
```

Comment:
In theory `NOT_APPLICABLE` just get filtered, support is here anyways just encase.